spec/user_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  subject do
+    User.create(name: 'Fatima', photo: 'https://unsplash.com/photos/F_-0BxGuVvo',
+                bio: 'I am a software engineer.', posts_counter: 0)
+  end
+  # Test most recent posts............
+
+  it 'returns the most recent 3 posts' do
+    subject.posts.create(title: 'Post 1', text: 'This is the first post.', comments_counter: 0, likes_counter: 0)
+    subject.posts.create(title: 'Post 2', text: 'This is the second post.', comments_counter: 0, likes_counter: 0)
+    subject.posts.create(title: 'Post 3', text: 'This is the third post.', comments_counter: 0, likes_counter: 0)
+    recent_posts = subject.recent_posts
+    expect(recent_posts.length).to eq(3)
+  end
```

Comment:
- [ ] @fatima-najafi Great work adding a test to check if the recent_posts method is working correctly, however giving it onlu3 posts won't effectively check if it's working correctly, kindly consider adding like 5 posts and check if it only displays 3 recent posts.